src/Microsoft.ML.Transforms/UngroupTransform.cs

```diff
@@ -179,21 +179,21 @@ public override bool CanShuffle
 
         protected override RowCursor GetRowCursorCore(Func<int, bool> predicate, Random rand = null)
         {
-            var activeInput = _schemaImpl.GetActiveInput(predicate);
+            var activeInput = _ungroupBinding.GetActiveInput(predicate);
             var inputCursor = Source.GetRowCursor(col => activeInput[col], null);
-            return new Cursor(Host, inputCursor, _schemaImpl, predicate);
+            return new Cursor(Host, inputCursor, _ungroupBinding, predicate);
         }
 
         public override RowCursor[] GetRowCursorSet(Func<int, bool> predicate,
             int n, Random rand = null)
         {
-            var activeInput = _schemaImpl.GetActiveInput(predicate);
+            var activeInput = _ungroupBinding.GetActiveInput(predicate);
             var inputCursors = Source.GetRowCursorSet(col => activeInput[col], n, null);
             return Utils.BuildArray<RowCursor>(inputCursors.Length,
-                x => new Cursor(Host, inputCursors[x], _schemaImpl, predicate));
+                x => new Cursor(Host, inputCursors[x], _ungroupBinding, predicate));
         }
 
-        private sealed class SchemaImpl : ISchema
+        private sealed class UngroupBinding
```

Comment:
>UngroupBinding [](start = 29, length = 14)

might be a good time to introduce a naming convention for these. No opinion on what it should be, as long as it is consistent :)